### PR TITLE
Get boot reason from property instead of cmdline.

### PR DESCRIPTION
**Pull request by jiaxuan-guo**
The old way of getting boot reason always return UNKNOWN. 
Use the latest way to get boot reason.

Reference: https://source.android.com/docs/core/architecture/bootloader/boot-reason

### CODE REVIEW

**Comment by jiaxuan-guo** [comment]
@jingdlu

**Comment by jingdlu** [comment]
> The old way of getting boot reason always return UNKNOWN. Use the latest way to get boot reason.
> 
> Reference: https://source.android.com/docs/core/architecture/bootloader/boot-reason

Suggest you to add this info to patch as detailed description.

File: crashlog/startupreason.c [hunk 1]
```diff
@@ -311,7 +311,8 @@ static void get_default_bootreason(char *bootreason) {
     unsigned int i;
     char bootreason_prop[PROPERTY_VALUE_MAX];
 
-    ret = get_cmdline_bootreason(bootreason_prop);
+    ret = property_get(PROP_BOOTREASON, bootreason_prop, "");
+    // ret = get_cmdline_bootreason(bootreason_prop);
```
Comment (jingdlu) [inline crashlog/startupreason.c]: We can remove this line. Just add one comment before "property_get" like "Use property_get starting from xxx version".

**Review by jingdlu — COMMENTED** [review]
It's better to capitalize first letter in patch title like "'Get".

**Review by jingdlu — COMMENTED** [review]
Everything is good except the patch title and description. Please consider the following title and description:

Get boot reason from property instead of cmdline

The old way of getting boot reason always returns UNKNOWN. This patch gets boot reason from property instead of cmdline.

**Comment by jiaxuan-guo** [comment]
> Everything is good except the patch title and description. Please consider the following title and description:
> 
> Get boot reason from property instead of cmdline
> 
> The old way of getting boot reason always returns UNKNOWN. This patch gets boot reason from property instead of cmdline.

Done. Thanks for your suggestion.

**Review by jingdlu — APPROVED** [review]

**Comment by jingdlu** [comment]
LGTM.

**Review by kwang13 — APPROVED** [review]

**Pull request closed without merging**